src/PioneerUtils.ts

```diff
@@ -256,7 +256,7 @@ export class PioneerUtils {
    */
   async chooseBranch() {
     const pioneerId = await this.getPioneerId();
-    const hashKey = `${this.config.studyName}/${this.getPioneerId()}`;
+    const hashKey = `${this.config.studyName}/${pioneerId}`;
```

Comment:
I have to admit, I'm not 💯 on what I'm doing, but it seemed wasteful to do the lookup twice (throwing away the first lookup). 
The `getPioneerId()` method isn't marked as `async`, so not sure why the line 258 has an `await`. In other places we used the value directly and didn't use `await`. So conversely, maybe just kill line 258 and the temporary `pioneerId` const.